### PR TITLE
internal/telemetry: don't send app-dependencies-loaded in the absence of Go deps

### PR DESCRIPTION
**Pull request by Julio-Guerra**
### What does this PR do?

Avoid sending the telemetry request `app-dependencies-loaded` when there is no Go build info available in the running Go program.
 

### Motivation

Datadog's Software Composition Analysis (SCA) relies on program deps, sent over `app-dependencies-loaded`, which must not be sent when they are missing.
This issue appeared with Bazel which doesn't seem to include Go build info (https://github.com/bazelbuild/rules_go/issues/3090).

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: internal/telemetry/telemetry_test.go [hunk 1]
```diff
@@ -151,7 +151,6 @@ func TestRegisterAppConfig(t *testing.T) {
 	require.Equal(t, RequestTypeAppStarted, req.RequestType)
 	appStarted := req.Payload.(*AppStarted)
 	cfg := appStarted.Configuration
-	require.Len(t, cfg, 2)
```
Comment (darccio) [inline internal/telemetry/telemetry_test.go:154]: @Julio-Guerra Is this change required?

Comment (Julio-Guerra) [reply internal/telemetry/telemetry_test.go:154]: yes, because I changed the global config with `dependencies_available` and `DD_TELEMETRY_DEPENDENCY_COLLECTION_ENABLED` reported by default all the times, so I removed the enforcement to 2 and let the following assertions fail anyway if the expected config values are missing.
